src/main/java/nearsoft/academy/bigdata/recommendation/MovieRecommender.java

```diff
@@ -33,54 +33,59 @@
-        Reader decoder = new InputStreamReader(gzipStream, "UTF-8");
-        BufferedReader buffered = new BufferedReader(decoder);
+    public void getData(){
+        try{
+            String line;
+            String idProduct="";
+            String idUser="";
+            String score;
+            InputStream fileStream = new FileInputStream(filename);
+            InputStream gzipStream = new GZIPInputStream(fileStream);
+            Reader decoder = new InputStreamReader(gzipStream, "UTF-8");
+            BufferedReader buffered = new BufferedReader(decoder);
 
-        while ((line = buffered.readLine()) != null){
-            if(line.indexOf("product/productId:") != -1){
-                reviews++;
-                idProduct = line.replace("product/productId: ","");
-                if(!productsTotal.containsKey(idProduct)){
-                    productsTotal.put(idProduct, totalProd);
-                    totalProd++;
+            while ((line = buffered.readLine()) != null){
+                if(line.indexOf("product/productId:") != -1){
+                    reviews++;
+                    idProduct = line.replace("product/productId: ","");
+                    if(!productsTotal.containsKey(idProduct)){
+                        productsTotal.put(idProduct, totalProd);
+                        totalProd++;
+                    }
                 }
-            }
-            if(line.indexOf("review/userId:") != -1){
-                idUser = line.replace("review/userId: ","");
-                if(!usersTotal.containsKey(idUser)){
-                    usersTotal.put(idUser, totalUsers);
-                    totalUsers++;
+                if(line.indexOf("review/userId:") != -1){
+                    idUser = line.replace("review/userId: ","");
+                    if(!usersTotal.containsKey(idUser)){
+                        usersTotal.put(idUser, totalUsers);
+                        totalUsers++;
+                    }
                 }
+                if(line.indexOf("review/score") != -1){
+                    score = line.replace("review/score: ","");
+                    scores.add(score);
+                    out.println(usersTotal.get(idUser) + "," +productsTotal.get(idProduct)+ "," + score);
+                }
+                
             }
-            if(line.indexOf("review/score") != -1){
-                score = line.replace("review/score: ","");
-                scores.add(score);
-                out.println(usersTotal.get(idUser) + "," +productsTotal.get(idProduct)+ "," + score);
-            }
-            
-        }
 
-        for (String idProd : productsTotal.keySet()) {
-            productsTotalReverse.put(productsTotal.get(idProd),idProd);
-        } 
-        out.close();
+            for (String idProd : productsTotal.keySet()) {
+                productsTotalReverse.put(productsTotal.get(idProd),idProd);
+            } 
+            out.close();
```

Comment:
I added it

Reply:
the `finally` means that even if the `try` fails, that part of the code will be executed. Why should the `for` loop be here?

Reply:
Thinking about it, I guess it shouldn't go there because it depends on what is done in the try.
It should only be "out.close ()"
Is it ok what I think?

Reply:
correct